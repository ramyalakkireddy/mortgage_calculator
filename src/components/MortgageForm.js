import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Grid, Button, Select, FormLabel, FormControl, MenuItem,
OutlinedInput, InputAdornment } from '@material-ui/core';
import mortgageImage from '../digital-mortgage.jpg';

const useStyles = makeStyles(theme => ({
   formControl: {
    marginTop: theme.spacing(1),
    width: '60%',
    minWidth: 120,
  },
  formLabel: {
    color: '#000000',
    fontSize: '1.15em',
    fontFamily: 'serif',
  }
}));

const amortizationPeriod = [...Array(60).keys()];
const interestTerm = [...Array(26).keys()];
const paymentFrequency = ['Monthly', 'Semi-Monthly', 'Bi-Weekly', 'Weekly', 'Accelerated Bi-Weekly', 'Accelerated Weekly'];

export default function MortgageForm(props) {
  const classes = useStyles();
  const { values, handleChange, handleCalculation } = props;

  return (
    <div>
    <Grid container spacing={3}>
      <Grid item xs={12} sm={6} md={6} lg={6}>
        <img src={mortgageImage} width="100%" height="100%" alt="mortgageHome" />
      </Grid>
      <Grid item xs={12} sm={6} md={6} lg={6}>
        <Grid container spacing={3} direction="column">
          <Grid item xs={12}>
            <FormLabel component="legend" className={classes.formLabel}>Mortgage Amount:</FormLabel>
            <FormControl className={classes.formControl} variant="outlined">
             <OutlinedInput
                id="outlined-adornment-amount"
                type="number"
                value={(values.mortgageAmount)}
                onChange={handleChange('mortgageAmount')}
                startAdornment={<InputAdornment position="start">$</InputAdornment>}
             />
            </FormControl>
          </Grid>
          <Grid item xs={12}>
            <FormLabel component="legend" className={classes.formLabel}>Amortization Period:</FormLabel>
              <FormControl variant="outlined" className={classes.formControl}>
                <Select
                  labelId="demo-simple-select-outlined-label"
                  id="demo-simple-select-outlined"
                  value={values.amortizationPeriod}
                  onChange={handleChange('amortizationPeriod')}
                >
                {amortizationPeriod.map((period) =>{
                    return (
                      <MenuItem value={period} key={period}>{period+' Years'}</MenuItem>
                      );
                    })
                }
                </Select>
              </FormControl>
          </Grid>
          <Grid item xs={12}>
            <FormControl variant="outlined" className={classes.formControl}>
             <FormLabel component="legend" className={classes.formLabel}>Payment Frequency:</FormLabel>
              <Select
                labelId="demo-simple-select-outlined-label"
                id="demo-simple-select-outlined"
                value={values.paymentFrequency}
                onChange={handleChange('paymentFrequency')}
              >
              {paymentFrequency.map((freq, index) =>{
                  return (
                    <MenuItem value={freq} key={index}>{freq}</MenuItem>
                    );
                  })
              }
              </Select>
            </FormControl>
          </Grid>
          <Grid item xs={12}>
             <FormLabel component="legend" className={classes.formLabel}>Interest Rate:</FormLabel>
             <FormControl className={classes.formControl} variant="outlined">
               <OutlinedInput
                  id="outlined-adornment-percentage"
                  type="number"
                  value={values.interestRate}
                  onChange={handleChange('interestRate')}
                  endAdornment={<InputAdornment position="end">%</InputAdornment>}
               />
             </FormControl>
          </Grid>
          <Grid item xs={12}>
            <FormLabel component="legend" className={classes.formLabel}>Interest Term:</FormLabel>
              <FormControl variant="outlined" className={classes.formControl}>
                <Select
                  labelId="demo-simple-select-outlined-label"
                  id="demo-simple-select-outlined"
                  value={values.interestTerm}
                  onChange={handleChange('interestTerm')}
                >
                {interestTerm.map((period) =>{
                    return (
                      <MenuItem value={period} key={period}>{period+' Years'}</MenuItem>
                      );
                    })
                }
                </Select>
              </FormControl>
          </Grid>
          <Grid item xs={6}>
            <Button color="primary" variant="contained" type="submit" onClick={handleCalculation('monthlyPayment')}>Calculate</Button>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
    </div>
  );
}
