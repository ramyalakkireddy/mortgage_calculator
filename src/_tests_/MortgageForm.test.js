import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MortgageForm from '../components/MortgageForm';

const values = {
  mortgageAmount: '100000',
  amortizationPeriod: '25',
  paymentFrequency: 'Monthly',
  interestRate: '5',
  interestTerm: '5'
}

const handleChange = jest.fn();
const handleCalculation = jest.fn();

describe("MortgageForm", () => {

  test("snapshot renders", () => {
    const component = renderer.create(<MortgageForm values={values} handleChange={handleChange} handleCalculation={handleCalculation} />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

});
