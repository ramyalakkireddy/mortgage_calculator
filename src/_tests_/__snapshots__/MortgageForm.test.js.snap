// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MortgageForm snapshot renders 1`] = `
<div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
    >
      <img
        alt="mortgageHome"
        height="100%"
        src="digital-mortgage.jpg"
        width="100%"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6 MuiGrid-grid-lg-6"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <legend
            className="MuiFormLabel-root makeStyles-formLabel-2"
          >
            Mortgage Amount:
          </legend>
          <div
            className="MuiFormControl-root makeStyles-formControl-1"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
              onClick={[Function]}
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  $
                </p>
              </div>
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
                disabled={false}
                id="outlined-adornment-amount"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value="100000"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-4"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <legend
            className="MuiFormLabel-root makeStyles-formLabel-2"
          >
            Amortization Period:
          </legend>
          <div
            className="MuiFormControl-root makeStyles-formControl-1"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="demo-simple-select-outlined"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                25 Years
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="25"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-4"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            className="MuiFormControl-root makeStyles-formControl-1"
          >
            <legend
              className="MuiFormLabel-root makeStyles-formLabel-2 MuiFormLabel-filled"
            >
              Payment Frequency:
            </legend>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="demo-simple-select-outlined"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Monthly
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="Monthly"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-4"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <legend
            className="MuiFormLabel-root makeStyles-formLabel-2"
          >
            Interest Rate:
          </legend>
          <div
            className="MuiFormControl-root makeStyles-formControl-1"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                id="outlined-adornment-percentage"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value="5"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  %
                </p>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-4"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <legend
            className="MuiFormLabel-root makeStyles-formLabel-2"
          >
            Interest Term:
          </legend>
          <div
            className="MuiFormControl-root makeStyles-formControl-1"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="demo-simple-select-outlined"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                5 Years
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="5"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-3 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-4"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Calculate
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
